refactor(accordion-item): migrate-to-fast-element (#311)
Vonage/vivid-3: libs/components/src/lib/accordion-item/README.md

# vwc-accordion-item

```js
<script type="module">
    import '@vonage/vivid/accordion-item';
</script>
```
## Heading
Add the `heading` attribute to set the heading text.

- Type: `string`
- Default: `''`
  
```html preview
  <vwc-accordion-item heading="accordion item with heading">
    <vwc-text font-face="body-1">
      Lorem Ipsum is simply dummy text of the printing and typesetting industry.
    </vwc-text>
  </vwc-accordion-item>
```

## Heading-Level

Use the `heading-level` attribute (or `headingLevel` property) to set the accordion heading level so it fits correctly within the outline of the page. Read more on [heading elements](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Heading_Elements).

- Type: `2` | `3` | `4` | `5` | `6`
- Default: `3`

> The following sets the heading of the accordion-item to level 2

Reviewer: ```suggestion
The following sets the heading of accordion-item to level 2
```

> ```html preview

Reviewer: ```suggestion
```html
```

<vwc-accordion-item heading="my heading" heading-level="2"></vwc-accordion-item>
```

which will output the shadow tree heading element to be rendered as a `<h2>` tag

e.g.

> ```js

Reviewer: ```suggestion
```html
```

    width: 500px;
  }
</style>
<div class="wrapper">
 <vwc-accordion-item heading="accordion item with metadata" meta="meta-data">
    <vwc-text font-face="body-1">
      Lorem Ipsum is simply dummy text of the printing and typesetting industry.
    </vwc-text>
  </vwc-accordion-item>
</div>
```
## Dense
You can turn the accordion-item into dense mode by adding the `dense` attribute.

- Type: `boolean`
- Default: `false`

```html preview
<style>
  .wrapper{
    width: 500px;
  }
</style>
<div class="wrapper">
 <vwc-accordion-item heading="accordion item with dense" dense>
    <vwc-text font-face="body-1">
      Lorem Ipsum is simply dummy text of the printing and typesetting industry.
    </vwc-text>
  </vwc-accordion-item>
</div>
```
## Icon
Add the `icon` attribute to add an icon to the heading. 

- Type: `string`
- Default: `''`

```html preview
<style>
  .wrapper{
    width: 500px;
  }
</style>
<div class="wrapper">
 <vwc-accordion-item heading="accordion item with icon" icon="chat-solid">
    <vwc-text font-face="body-1">
      Lorem Ipsum is simply dummy text of the printing and typesetting industry.
    </vwc-text>
  </vwc-accordion-item>
</div>
```

## Icon-Trailing

> Add the `icon-trailing` attribute to add an icon to the right of the heading text.  Mind that `icon-trailing` will overide the Indicator.

Reviewer: ```suggestion
Add the `icon-trailing` attribute to add an icon to the right of the heading text.  Mind that `icon-trailing` will override the Indicator.
```

- Type: `boolean`
- Default: `false`

```html preview
<style>
  .wrapper{
    width: 500px;
  }
</style>
<div class="wrapper">
 <vwc-accordion-item heading="accordion item with icon-trailing" icon="chat-solid" icon-trailing>
    <vwc-text font-face="body-1">
      Lorem Ipsum is simply dummy text of the printing and typesetting industry.
    </vwc-text>
  </vwc-accordion-item>
</div>
```

  ## Methods

| Method       | Type       | Description                                      |
| ------------ | ---------- | ------------------------------------------------ |
| `hide`       | `(): void` | Closes the accordion item from the open state.  |

> | `show`       | `(): void` | Opens the accordion item from the closed state. |

Reviewer: I started aligning methods to our members convention. there shouldn't be any difference I think. consider? or think it should be like this?

see event context
https://vivid-dy25ra84f90g.deno.dev/components/calendar

Author: It is more clear this way. Perhaps we can discuss it further. It isn't relevant for this PR since all the components need to change.

Reviewer: right but this is a good place to start

# vwc-accordion-item

```js
<script type="module">
    import '@vonage/vivid/accordion-item';
</script>
```
## Heading
Add the `heading` attribute to set the heading text.

- Type: `string`
- Default: `''`
  
```html preview
<style>
  .wrapper{
    width: 500px;
  }
</style>
<div class="wrapper">
  <vwc-accordion-item heading="accordion item with heading">
    <vwc-text font-face="body-1">
      Lorem Ipsum is simply dummy text of the printing and typesetting industry.
    </vwc-text>
  </vwc-accordion-item>
</div>
```

## Heading-Level

> Use the `heading-level` attribute to change the accordion item's size.

Reviewer 2: ```suggestion
Use the `heading-level` attribute to change the accordion heading to fit the page hierarchy.```

- Type: `2` | `3` | `4` | `5` | `6`
- Default: `3`

> ```js

Reviewer: ```suggestion
```html preview
```

Author: It's not a code example, so it does not have a preview

Reviewer: doesn't have to annotate with _preview_. but it's an _html_ and not _js_

Reviewer: I'm not sure it's very informative for readers. they dont have interior knowledge of the template.
maybe drop in a quote from accordion aria practice that explains the heading hierarchy.
add a snippet preview with saying something like "the following will output..."

Reviewer: ```suggestion
```html preview
```

</h4>
<h5 class="accordion-item-header">
	<button class="accordion-item-button"></button>
</h5>
<h6 class="accordion-item-header">
	<button class="accordion-item-button"></button>
</h6>
```
## Open
Use the `open` attribute to indicate whether the accordion item is open.
Alternatively, you can use the `show()` and `hide()` methods as well as `toggleOpen()`.
- Type: `boolean`
- Default: `false`

```html preview
<style>
  .wrapper{
    width: 500px;
  }
</style>
<div class="wrapper">
 <vwc-accordion-item heading="Click to toggle accordion item" open>
    <vwc-text font-face="body-1">
      Lorem Ipsum is simply dummy text of the printing and typesetting industry.
    </vwc-text>
  </vwc-accordion-item>
</div>
```
## No-Indicator
Add the `no-indicator` attribute to remove the indicator icon from the heading.

- Type: `boolean`
- Default: `false`

```html preview
<style>
  .wrapper{
    width: 500px;
  }
</style>
<div class="wrapper">
 <vwc-accordion-item heading="accordion item without indicator" no-indicator>
    <vwc-text font-face="body-1">
      Lorem Ipsum is simply dummy text of the printing and typesetting industry.
    </vwc-text>
  </vwc-accordion-item>
</div>
```

> ## Icon

Reviewer: icon is a common member. sort to the end (icon-trailing too)

Add the `icon` attribute to add an icon to the heading. 

- Type: `string`
- Default: `''`

```html preview
<style>
  .wrapper{
    width: 500px;
  }
</style>
<div class="wrapper">
 <vwc-accordion-item heading="accordion item with Icon" icon="chat-solid">
    <vwc-text font-face="body-1">
      Lorem Ipsum is simply dummy text of the printing and typesetting industry.
    </vwc-text>
  </vwc-accordion-item>
</div>
```

## Icon-Trailing

> Add the `icon-trailing` attribute to add an icon to the right of the heading text.

Reviewer 2: ```suggestion
Mind that `icon-trailing` will overide the Indicator```

- Type: `boolean`
- Default: `false`

```html preview
<style>
  .wrapper{
    width: 500px;
  }
</style>
<div class="wrapper">
 <vwc-accordion-item heading="accordion item with Icon-Trailing" icon="chat-solid" icon-trailing>
    <vwc-text font-face="body-1">
      Lorem Ipsum is simply dummy text of the printing and typesetting industry.
    </vwc-text>
  </vwc-accordion-item>
</div>
```

## Meta
Add the `meta` attribute to add metadata to the heading. 

- Type: `string`
- Default: `''`

```html preview
<style>
  .wrapper{
    width: 500px;
  }
</style>
<div class="wrapper">
 <vwc-accordion-item heading="accordion item with Metadata" meta="meta-data">
    <vwc-text font-face="body-1">
      Lorem Ipsum is simply dummy text of the printing and typesetting industry.
    </vwc-text>
  </vwc-accordion-item>
</div>
```

> ## Dense

Reviewer: let's make sure this is still required @jshenkman

# vwc-accordion-item

```js
<script type="module">
    import '@vonage/vivid/accordion-item';
</script>
```
## Heading
Add the `heading` attribute to set the heading text.

- Type: `string`
- Default: `''`
  
```html preview
<style>
  .wrapper{
    width: 500px;
  }
</style>
<div class="wrapper">
  <vwc-accordion-item heading="accordion item with heading">
    <vwc-text font-face="body-1">
      Lorem Ipsum is simply dummy text of the printing and typesetting industry.
    </vwc-text>
  </vwc-accordion-item>
</div>
```

## Heading-Level
Use the `heading-level` attribute to change the accordion heading to fit the page hierarchy.

- Type: `2` | `3` | `4` | `5` | `6`
- Default: `3`

```js
<h2 class="header">
	<button class="button"></button>
</h2>
<h3 class="header">
	<button class="button"></button>
</h3>
<h4 class="header">
	<button class="button"></button>
</h4>
<h5 class="header">
	<button class="button"></button>
</h5>
<h6 class="header">
	<button class="button"></button>
</h6>
```
## Open

> Use the `open` attribute to indicate whether the accordion item is open.

Reviewer: ```suggestion
Use the `open` attribute to set the accordion-item's open state.
```

Alternatively, you can use the [`show()`](#show) and [`hide()`](#hide) methods.
- Type: `boolean`
- Default: `false`

```html preview
<style>
  .wrapper{
    width: 500px;
  }
</style>

> <div class="wrapper">

Reviewer: this adds more reading material. remove from all?

Author: @yinonov but then the indicator is far on the right of the example.

 <vwc-accordion-item heading="Click to toggle accordion item" open>
    <vwc-text font-face="body-1">
      Lorem Ipsum is simply dummy text of the printing and typesetting industry.
    </vwc-text>
  </vwc-accordion-item>
</div>
```
## No-Indicator
Add the `no-indicator` attribute to remove the indicator icon from the heading.

- Type: `boolean`
- Default: `false`

```html preview
<style>
  .wrapper{
    width: 500px;
  }
</style>
<div class="wrapper">
 <vwc-accordion-item heading="accordion item without indicator" no-indicator>
    <vwc-text font-face="body-1">
      Lorem Ipsum is simply dummy text of the printing and typesetting industry.
    </vwc-text>
  </vwc-accordion-item>
</div>
```
## Meta
Add the `meta` attribute to add metadata to the heading. 

- Type: `string`
- Default: `''`

```html preview
<style>
  .wrapper{
    width: 500px;
  }
</style>
<div class="wrapper">
 <vwc-accordion-item heading="accordion item with metadata" meta="meta-data">
    <vwc-text font-face="body-1">
      Lorem Ipsum is simply dummy text of the printing and typesetting industry.
    </vwc-text>
  </vwc-accordion-item>
</div>
```

> ## Dense

Reviewer: still need to discuss (preferably omit). maybe comment out first?

      Lorem Ipsum is simply dummy text of the printing and typesetting industry.
    </vwc-text>
  </vwc-accordion-item>
</div>
```
## Icon
Add the `icon` attribute to add an icon to the heading. 

- Type: `string`
- Default: `''`

```html preview
<style>
  .wrapper{
    width: 500px;
  }
</style>
<div class="wrapper">
 <vwc-accordion-item heading="accordion item with icon" icon="chat-solid">
    <vwc-text font-face="body-1">
      Lorem Ipsum is simply dummy text of the printing and typesetting industry.
    </vwc-text>
  </vwc-accordion-item>
</div>
```

## Icon-Trailing
Add the `icon-trailing` attribute to add an icon to the right of the heading text.  Mind that `icon-trailing` will override the Indicator.

- Type: `boolean`
- Default: `false`

```html preview
<style>
  .wrapper{
    width: 500px;
  }
</style>
<div class="wrapper">
 <vwc-accordion-item heading="accordion item with icon-trailing" icon="chat-solid" icon-trailing>
    <vwc-text font-face="body-1">
      Lorem Ipsum is simply dummy text of the printing and typesetting industry.
    </vwc-text>
  </vwc-accordion-item>
</div>
```

## Methods

### Show

- Type: function
- Returns: `void`

> Opens the accordion item.

Reviewer: should be ordered above the type & default

### Hide

- Type: function
- Returns: `void`

> Closes the accordion item.

Reviewer: why do we have these methods anyway? I think it's a good opportunity to try and question these. are they redundant? do they add some capability which lacks in a property member? maybe it should accept callback and that's its purpose?

Author: @yinonov My opinion is that we should remove all the hide and show methods.. from the side-drawer and from other components as well.

Reviewer: go ahead
